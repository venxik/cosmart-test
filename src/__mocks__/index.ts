export * from './wrapper';
export * from './responseMock';
