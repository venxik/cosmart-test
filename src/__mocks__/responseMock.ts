export const getBooksResponseMock = {
  key: '/subjects/love',
  name: 'love',
  subject_type: 'subject',
  work_count: 15853,
  works: [
    {
      authors: [{ key: '/authors/OL4327048A', name: 'Emily BrontÃ«' }],
      key: 'testing',
      title: 'testing',
      edition_count: 123,
      has_fulltext: true,
      ia: 'testing',
      bookingDate: '1234',
    },
  ],
};
