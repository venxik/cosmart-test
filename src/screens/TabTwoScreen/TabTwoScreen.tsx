import { StyleSheet } from 'react-native';

import { BookedItemList, View } from '../../components';

export default function TabTwoScreen() {
  return (
    <View style={styles.container}>
      <BookedItemList />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
