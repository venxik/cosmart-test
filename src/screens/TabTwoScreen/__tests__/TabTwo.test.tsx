import { screen } from '@testing-library/react-native';
import { renderWithProviders } from '../../../__mocks__';
import TabTwoScreen from '../TabTwoScreen';

describe('TabTwoScreen', () => {
  it('match snapshot', async () => {
    renderWithProviders(<TabTwoScreen />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
