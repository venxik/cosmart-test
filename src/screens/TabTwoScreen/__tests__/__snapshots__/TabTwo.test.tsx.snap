// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabTwoScreen match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#2f95dc",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#2f95dc",
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="empty_container"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "fontSize": 45,
                "fontWeight": "700",
              },
            ]
          }
        >
          Booking List is Empty
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
